qwed_tax/guards/capital_gains_guard.py

```
@@ -10,27 +10,29 @@ class CapitalGainsGuard:
         """
         Calculates Holding Period in days and returns 'LTCG' or 'STCG'.
+        Raises ValueError on unparseable dates or unknown asset types.
         """
         try:
             d1 = datetime.strptime(purchase_date, "%Y-%m-%d")
             d2 = datetime.strptime(sale_date, "%Y-%m-%d")
             days = (d2 - d1).days
         except ValueError:
-            return "ERROR_DATE_FORMAT"
-        
+            raise ValueError(f"Invalid date format. Expected YYYY-MM-DD, got '{purchase_date}' and '{sale_date}'.")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)